### PR TITLE
feat: upgrade bones and use `iroh` based multiplayer

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
the clippy failures seem unrelated, let me know if you want me to fix them here

File: src/ui/network_game.rs [hunk 1]
```diff
@@ -456,7 +459,8 @@ pub fn network_game_menu(
                                 .clicked()
                                 {
                                     *status = NetworkGameStatus::Searching;
-                                    online::start_search_for_game(matchmaking_server.clone(), *player_count);
+                                    let server = matchmaking_server.parse().expect("invalid server id");
```
Comment (dignifiedquire) [inline src/ui/network_game.rs:462]: this should really not panic but show an error in the UI

Comment (dignifiedquire) [reply src/ui/network_game.rs:462]: but I don't exactly know how, so I'll leave that up to you ;)

Comment (MaxCWhitehead) [reply src/ui/network_game.rs:462]: yeah that's fine - we don't have great usage of friendly error display yet to model around.

File: Cargo.toml [hunk 2]
```diff
@@ -152,6 +152,6 @@ ignored = [
 
 # # Uncomment for testing during bones development, if you clone bones adjacent to the jumpy
 # # repo.
-# [patch.'https://github.com/fishfolk/bones']
+[patch.'https://github.com/fishfolk/bones']
```
Comment (MaxCWhitehead) [inline Cargo.toml]: Super minor comment but could we remove this diff / keep this line commented? :) Thanks

Otherwise I think this looks good to merge.

**Comment by MaxCWhitehead** [comment]
I setup a matchmaker with persistent node id and changed jumpy's default to this: `lokmcpce2uzezmh2iq3vf23cbu3grzrct66qcpdfrizmcrrlk5uq`

@zicklag if we have any reason want to use existing host can do that as well just throwing something in here for now.

File: Cargo.toml (hunk 2 above)
Comment (MaxCWhitehead) [reply Cargo.toml]: fixed this

**Review by MaxCWhitehead — APPROVED** [review]

**Pull request merged**